### PR TITLE
Compress HTTP service responses with gzip

### LINKED ISSUE #1532: Compress HTTP service responses
The JSON responses could be gzipped or brotlified.

### PR DESCRIPTION
**Pull request by rajeshkumargp**
Compress reponses whose status_code is 2XX

Fixes #1532

### CODE REVIEW

File: http_service/bugbug_http/app.py [hunk 1]
```diff
@@ -284,6 +285,28 @@ def get_result(job):
...
+    :return: response with gzip compressed data
+    :rtype: Response
+    """
+
+    gzip_buffer = gzip.compress(orjson.dumps(data), compresslevel=9)
+
+    response = Response(status=status_code)
+    response.set_data(gzip_buffer)
+    response.headers["Content-Encoding"] = "gzip"
+    response.headers["Content-Length"] = len(response.get_data())
```
Comment (marco-c) [inline http_service/bugbug_http/app.py:304]: Nit: You can probably just do:
```suggestion
    response.headers["Content-Length"] = len(gzip_buffer)
```

Comment (rajeshkumargp) [reply http_service/bugbug_http/app.py]: Fixed

**Comment by rajeshkumargp** [comment]
While running `python -m pytest http_service/tests/`,  we are getting responses (as in https://github.com/mozilla/bugbug/blob/master/http_service/tests/test_bug_classification.py#L29), which belongs to <class 'flask.wrappers.Response'>.  
When using `requests` for getting reponses, we will get `<class 'requests.models.Response'>` object. 

The object of `<class 'requests.models.Response'>` decompresses automatically, and retrieval of  gzipped responses is fine. Please refer https://github.com/mozilla/bugbug/pull/1632#discussion_r438336116 

Accordingly, the test cases referring `json` conversion is changed.

https://github.com/mozilla/bugbug/blob/master/http_service/tests/test_bug_classification.py#L31
https://github.com/mozilla/bugbug/blob/master/http_service/tests/test_bug_classification.py#L36
https://github.com/mozilla/bugbug/blob/master/http_service/tests/test_bug_classification.py#L45

https://github.com/mozilla/bugbug/blob/master/http_service/tests/test_push_schedules.py#L32


In other test responses, gzip compression is not done, So the test cases were failed earlier because they are different from responses of `<class 'flask.wrappers.Response'>` is different from <class 'requests.models.Response'>` object, when `response.json`  is called.

The code is updated now.

File: http_service/bugbug_http/app.py (hunk 1 above)
Comment (rajeshkumargp) [reply http_service/bugbug_http/app.py]: Resolved

**Comment by marco-c** [comment]
@rajeshkumargp you might be interested in #1616, it is kind of similar ;)

**Comment by rajeshkumargp** [comment]
> @rajeshkumargp you might be interested in #1616, it is kind of similar ;)

Sure, @marco-c

**Pull request merged**